File: src/main/java/macro303/keschet/pieces/Thief.java
package macro303.keschet.pieces;

import macro303.keschet.Colour;
import macro303.keschet.Direction;
import org.jetbrains.annotations.NotNull;

import static macro303.keschet.Direction.*;

/**
 * Created by Macro303 on 2018-02-09.
 */
public class Thief extends Piece {
	@NotNull
//	private static final String symbol = "\uD83D\uDDE1";
	private static final String symbol = "T";
	private static final int maxDistance = 1;
	@NotNull
	private static final Direction[] validDirections = new Direction[]{NORTH, NORTH_EAST, EAST, SOUTH_EAST, SOUTH, SOUTH_WEST, WEST, NORTH_WEST};

	public Thief(@NotNull Colour teamColour) {
		super(teamColour, maxDistance, symbol, validDirections);
	}

	public Thief() {
		super(maxDistance, symbol, validDirections);
	}

	@Override
	public String toString() {
		return "Thief{}";
	}
}

File: src/main/java/macro303/keschet/pieces/Thief.kt
package macro303.keschet.pieces

import macro303.keschet.Console
import macro303.keschet.Direction

class Thief(teamColour: Console.Colour) : Piece(teamColour = teamColour, maxDistance = 1, symbol = "T", validDirections = arrayOf(Direction.NORTH, Direction.NORTH_EAST, Direction.EAST, Direction.SOUTH_EAST, Direction.SOUTH, Direction.SOUTH_WEST, Direction.WEST, Direction.NORTH_WEST)) {

	override fun equals(other: Any?): Boolean {
		if (this === other) return true
		if (other !is Thief) return false
		if (!super.equals(other)) return false
		return true
	}

	override fun hashCode(): Int {
		return super.hashCode()
	}

	override fun toString(): String {
		return "Thief() ${super.toString()}"
	}
}